packages/vscode-extension/src/debugging/DebugAdapter.ts

```diff
@@ -150,7 +153,22 @@ export class DebugAdapter extends DebugSession {
             const decodedData = Buffer.from(base64Data, "base64").toString("utf-8");
             const sourceMap = JSON.parse(decodedData);
             const consumer = await new SourceMapConsumer(sourceMap);
-            this.sourceMaps.push([message.params.url, message.params.scriptId, consumer]);
+
+            // This line is here because of a problem with sourcemaps for expo projects,
+            // that was addressed in this PR https://github.com/expo/expo/pull/29463,
+            // unfortunately it still requires changes to metro that were attempted here
+            // https://github.com/facebook/metro/pull/1284 we should monitor the situation
+            // in upcoming versions and if the changes are still not added bump the version below.
+            // more over offset should only be applied to the main bundle sourcemap as other files
+            // (generated during reload) do not contain the prelude causing the issue
+            const shouldApplyOffset =
+              semver.lte(getReactNativeVersion(), "0.76.0") && this.sourceMaps.length === 0;
+            this.sourceMaps.push([
+              message.params.url,
+              message.params.scriptId,
+              consumer,
+              shouldApplyOffset ? this.lineOffset : 0,
```

Comment:
this seems to assume that we will get some console log call before source maps are provided such that this.lineOffect can be set.